elf_binary.cc

```diff
@@ -296,11 +296,163 @@ void ELFBinary::ParsePhdrs() {
             ParseDynamic(phdr->p_offset, phdr->p_filesz);
         } else if (phdr->p_type == PT_INTERP) {
             LOG(INFO) << "Found PT_INTERP.";
+        } else if (phdr->p_type == PT_GNU_EH_FRAME) {
+            ParseEHFrameHeader(phdr->p_offset, phdr->p_filesz);
         }
     }
     CHECK(!phdrs_.empty());
 }
 
+void ELFBinary::ParseEHFrameHeader(size_t off, size_t size) {
+    const char* const efh_base = head_ + off;
+    int efh_offset = 0;
+
+    eh_frame_header_.version = *(efh_base + efh_offset);
```

Comment:
This `x = ptr[off]; off += sizeof(x); ...` pattern appears often when parsing binaries. I think this is a bit error prone. How about introducing a helper function. E.g.,

```
#include <stdio.h>
#include <string.h>

struct S {
    char c;
    int i;
    char d;
};

int main() {
    const char* msg = "123456789";
    int off = 0;
    auto read = [msg, &off](auto* p) {
        memcpy(p, msg + off, sizeof(*p));
        off += sizeof(*p);
    };

    S s;
    read(&s.c);
    read(&s.i);
    read(&s.d);
    printf("%c %c\n", s.c, s.d);
}

```

Reply:
It looks good. I will introduce it.